### PR TITLE
Optional singleton

### PR DESCRIPTION
**Pull request by christophwelling**
Addresses #266 
No you can pass create_new=True to a Singleton's __call__ method to explicitly create a new class instance

### CODE REVIEW

File: NuRadioReco/autodoc/pages/utilities.rst [hunk 1]
```diff
@@ -31,4 +31,19 @@ domain. The functions are wrappers around the the numpy real Fourier transform
...
+Singleton
+^^^^^^^^^^^^
+For some classes (for example, the Detector class), it is usually a good idea to only have
+one instance of it active at a time. This can be accomplished by assigning the Singleton
+metaclass to it. The metaclass supercedes the __call__ method
+(e.g. :code:`NuRadioReco.detector.detector.Detector()`) to check if an instance of the class
+already exists and return that instance instead of creating a new one.
+
+If you want to enforce the creation of a new class instance, you can overwrite this behavior
+by passing create_new=True as a parameter to the __call__ method.
```
Comment (clark2668) [inline NuRadioReco/autodoc/pages/utilities.rst:49]: is the create_new list as a param somewhere in the singleton class definition? Might be good if it were.

**Review by clark2668 — COMMENTED** [review]
I made one comment about adding the create_new parameter to the class def, but otherwise I think this looks good, and can confirm it works in the case I was having trouble with.

File: NuRadioReco/autodoc/pages/utilities.rst (hunk 1 above)
Comment (christophwelling) [reply NuRadioReco/autodoc/pages/utilities.rst:49]: Yes, it's in the docstring of the metaclass as well

**Comment by cg-laser** [comment]
I think it is not clear yet how to call a new detector instance (at least not to me). Can you provide an explicit example? And ideally also add it to the docstring of the detector class as this is the point where people will look first?

**Comment by christophwelling** [comment]
> so e.g. `det = Detector(source='json', create_new=True)` will do the job?

Yes.  I also noticed that by default the __init__ is not listed in the rendered documentation. I made a commit to change that for all classes.

**Comment by cg-laser** [comment]
I think this is good to go but can you add the new parameter to the docstring of the detector __init__ function. I think otherwise it will be very difficult to see that this parameter exists.

**Comment by clark2668** [comment]
Just wanted to comment that I started using this, and it was very helpful. Also looks good to merge after you're happy with the docs. Thanks!

**Comment by christophwelling** [comment]
Then can someone approve this, so we stick to the process?

**Review by anelles — APPROVED** [review]
Looks good.

**Review by cg-laser — CHANGES_REQUESTED** [review]
you did not add the description of the new __init__ parameter directly to the init function of the detector (see my comment/request above)

**Comment by christophwelling** [comment]
I added it. Looks like sphinx does not know how to handle metaclasses, that's a bit unfortunate.

**Review by cg-laser — APPROVED** [review]
thanks

**Pull request merged**